torch_xla/csrc/ops/tensor_data.cpp

```diff
@@ -59,23 +59,27 @@ TensorData::TensorData(at::Tensor tensor, Device device)
 
 std::string TensorData::ToString() const {
   std::stringstream ss;
-  ss << Node::ToString() << ", device=" << device_ << ", values=";
-  PrintTensorValues(ss, tensor_);
+  ss << Node::ToString() << ", device=" << device_;
+  if (ShouldMakeConstant(tensor_)) {
+    ss << ", values=";
+    PrintTensorValues(ss, tensor_);
+  }
   return ss.str();
 }
 
 XlaOpVector TensorData::Lower(LoweringContext* loctx) const {
   if (ShouldMakeConstant(tensor_)) {
-    xla::ComputationClient::DataPtr data = TensorToXlaData(tensor_, device_);
-    return ReturnOp(loctx->GetParameter(data), loctx);
+    xla::Literal value = GetTensorLiteral(tensor_, &shape(), &device_);
+    return ReturnOp(xla::ConstantLiteral(loctx->builder(), value), loctx);
   }
-  xla::Literal value = GetTensorLiteral(tensor_, &shape(), &device_);
-  return ReturnOp(xla::ConstantLiteral(loctx->builder(), value), loctx);
+  xla::ComputationClient::DataPtr data = TensorToXlaData(tensor_, device_);
+  return ReturnOp(loctx->GetParameter(data), loctx);
 }
 
 bool TensorData::ShouldMakeConstant(const at::Tensor& tensor) {
```

Comment:
We've reversed the semantics of this method and the name stayed the same. Was this wrong before? If not, I'd consider a different name for it.

Reply:
Was wrong before. Used to be ShouldBeOnDevice(), but then I changed the name, but not reflected the implementation.